examples/timebase_master.py

```diff
@@ -0,0 +1,148 @@
+            print("New pos:", jack.position2dict(pos))
+
+        # Adapted from:
+        # https://github.com/jackaudio/jack2/blob/develop/example-clients/transport.c#L66
+        if new_pos:
+            pos.beats_per_bar = self.beats_per_bar
+            pos.beats_per_minute = self.bpm
+            pos.beat_type = self.beat_type
+            pos.ticks_per_beat = self.ticks_per_beat
+            pos.valid |= jack._lib.JackPositionBBT
+
+            minutes = pos.frame / (pos.frame_rate * 60.0)
+            abs_tick = minutes * pos.beats_per_minute * pos.ticks_per_beat
+            abs_beat = abs_tick / self.ticks_per_beat
+
+            pos.bar = int(abs_beat / pos.beats_per_bar)
+            pos.beat = int(abs_beat - (pos.bar * pos.beats_per_bar) + 1)
+            pos.tick = int(abs_tick - (abs_beat * pos.ticks_per_beat))
+            pos.bar_start_tick = pos.bar * pos.beats_per_bar * pos.ticks_per_beat
+            pos.bar += 1  # adjust start to bar 1
+        else:
+            # Compute BBT info based on previous period.
+            pos.tick += int(nframes * pos.ticks_per_beat *
+                            pos.beats_per_minute / (pos.frame_rate * 60))
+
+            while pos.tick >= pos.ticks_per_beat:
+                pos.tick -= int(pos.ticks_per_beat)
+                pos.beat += 1
+
+                if pos.beat > pos.beats_per_bar:
+                    pos.beat = 1
+                    pos.bar += 1
+                    pos.bar_start_tick += pos.beats_per_bar * pos.ticks_per_beat
+
+                    if self.debug:
+                        print("Pos:", jack.position2dict(pos))
+
+    def become_timebase_master(self, conditional=None):
+        return self.set_timebase_callback(self._tb_callback, conditional
+                                          if conditional is not None
+                                          else self.conditional)
+
+
+def main(args=None):
+    ap = argparse.ArgumentParser(description=__doc__)
+    ap.add_argument(
+        '-d', '--debug',
+        action='store_true',
+        help="Enable debug messages")
+    ap.add_argument(
+        '-c', '--conditional',
+        action='store_true',
+        help="Exit if another timebase master is already active")
+    ap.add_argument(
+        '-n', '--client-name',
+        metavar='NAME',
+        default='timebase',
+        help="JACK client name (default: %(default)s)")
+    ap.add_argument(
+        '-m', '--meter',
+        default='4/4',
+        help="Meter as <beats-per-bar>/<beat-type> (default: %(default)s)")
+    ap.add_argument(
+        '-t', '--ticks-per-beat',
+        type=int,
+        metavar='NUM',
+        default=1920,
+        help="Ticks per beat (default: %(default)s)")
+    ap.add_argument(
+        'tempo',
+        nargs='?',
+        type=float,
+        default=120.0,
+        help="Tempo in beats per minute (default: %(default)s)")
+
+    args = ap.parse_args(args if args is not None else sys.argv[1:])
```

Comment:
I think you are doing what Python would be doing anyway.

https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args

I think you can just pass `args` and be done with it.

Reply:
Old habit. I don't think `argparse` always did that. But in Python 3 it should always work. Fixed in 60e57a89d78e329b247a80788f70ae8271c97db2.